### PR TITLE
Add Custom GroupChat Speaker Selection to tutorial

### PR DESCRIPTION
**Pull request by kevin666aa**
## Why are these changes needed?
- Add custom speaker selection to tutorial
- update research about stateflow

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: website/docs/Research.md [hunk 1]
```diff
@@ -73,3 +73,14 @@ For technical details, please check our technical report and research publicatio
       primaryClass={cs.AI}
 }
 ```
+
+* [StateFlow: Enhancing LLM Task-Solving through State-Driven Workflows]https://arxiv.org/abs/2403.11322. Yiran Wu, Tianwei Yue, Shaokun Zhang, Chi Wang, Qingyun Wu. ArXiv preprint arXiv:2403.11322 (2024).
+
+@misc{wu2024stateflow,
```
Comment (skzhang1) [inline website/docs/Research.md:79]:
```bibtex

Comment (skzhang1) [reply website/docs/Research.md:79]: need ``` bibtext

**Review by jackgerrits — APPROVED** [review]

File: website/docs/tutorial/conversation-patterns.ipynb
Comment (ekzhu) [inline website/docs/tutorial/conversation-patterns.ipynb:1]: Can you create a new topic category under /website/docs/topics called `groupchat`? Then, you can create a notebook there called `customized_speaker_selection.ipynb`. Use the `non-openai-models` category as a reference. 

We don't want to clutter the tutorial chapter with too many details.

**Review by ekzhu — CHANGES_REQUESTED** [review]
Move the documentation to a topic page.

File: website/docs/tutorial/conversation-patterns.ipynb
Comment (ekzhu) [reply website/docs/tutorial/conversation-patterns.ipynb:1]: #2243

File: website/docs/tutorial/conversation-patterns.ipynb [hunk 2]
```diff
@@ -1559,6 +1559,496 @@
...
+    "```\n",
+    "\n",
+    "We define the following agents:\n",
+    "\n",
+    "- Initializer: Start the workflow by sending a task.\n",
+    "- Coder: Retrieve papers from the internet by writing code.\n",
+    "- Executor: Execute the code.\n",
+    "- Scientist: Read the papers and write a summary.\n",
+    "\n",
+    "In the Figure, we define a simple workflow for research with 4 states: Init, Retrieve, Reserach and End. Within each state, we will call different agents to perform the tasks.\n",
```
Comment (joshkyh) [inline website/docs/tutorial/conversation-patterns.ipynb:1609]:
```suggestion
    "In the Figure, we define a simple workflow for research with 4 states: Init, Retrieve, Research and End. Within each state, we will call different agents to perform the tasks.\n",
```

**Review by joshkyh — APPROVED** [review]
Thanks for tutorial! :)

File: website/docs/tutorial/conversation-patterns.ipynb
Comment (kevin666aa) [reply website/docs/tutorial/conversation-patterns.ipynb:1]: @ekzhu I already created the category for this. Can you check it out?

I think we should also navigate some docs from tutorial here? 
I agree that the tutorial is way too long. We can introduce the very basic of group chat there, and direct to this topic category for more advanced features such as nested chat, custom speaker function, etc.

Comment (ekzhu) [reply website/docs/tutorial/conversation-patterns.ipynb:1]: Yes you are right. We can do this as part of #2243

File: website/docs/topics/groupchat/_category_.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "position": 5,
+    "label": "Using GroupChat",
```
Comment (ekzhu) [inline website/docs/topics/groupchat/_category_.json:3]: Just "GroupChat" is enough.

File: website/docs/topics/groupchat/customized_speaker_selection.ipynb
Comment (ekzhu) [inline website/docs/topics/groupchat/customized_speaker_selection.ipynb:1]: This block 

```python
config_list = autogen.config_list_from_json(
    "OAI_CONFIG_LIST",
    filter_dict={
        "model": ["gpt-4-1106-preview"],
    },
)
```

Use a raw python list of dictionary for the config_list. This simplies the set up for notebooks and make it easy to run in codelab. Use os.getenv("...") to get the environment variable for open ai keys.

**Review by ekzhu — CHANGES_REQUESTED** [review]

**Review by ekzhu — APPROVED** [review]

**Pull request merged**